agent/submatview/store.go

```diff
@@ -221,7 +221,7 @@ func (s *Store) readEntry(req Request) (string, Materializer, error) {
 		return "", nil, err
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
For some reason, this change is breaking `TestServiceMultipleTagsWatch` in `api/watch`. Still investigating why.